vpr/src/pack/cluster.cpp

```diff
@@ -2050,6 +2062,13 @@ static void update_total_gain(float alpha, float beta, bool timing_driven, bool
                 cur_pb->pb_stats->sharinggain[blk_id] = 0;
             }
 
+            //Attraction Group stuff
+            AttractGroupId atom_grp_id = attraction_info.get_atom_attraction_group(blk_id);
+            if (atom_grp_id != NO_ATTRACTION_GROUP && atom_grp_id == cluster_att_grp_id) {
+                //increase gain of atom by 1
+                cur_pb->pb_stats->gain[blk_id]++;
```

Comment:
use the attraction group gain to update

Reply:
right now you are updating the blocks that are part of the attraction group multiple times - decide if that is a problem. If it isn't a problem, comment that that is what you are doing. If it is a problem, write code to make sure that you only do this once (i.e. did the cluster attraction group just change) - if you can do this it is good because it can be faster